docs/src/simulation_tips.md

```diff
@@ -93,7 +93,7 @@ always work on CPUs, but when their complexity is high (in terms of number of ab
 the compiler can't translate them into GPU code and they fail for GPU runs. (This limitation is discussed 
 in [this Github issue](https://github.com/CliMA/Oceananigans.jl/issues/1241) and contributors are welcome.)
 For example, in the example below, calculating `u²` works in both CPUs and GPUs, but calculating 
-`KE` only works on CPUs:
+`KE` may not work in some GPUs:
```

Comment:
Hmm, I think it works now --- at least, we test it.

Reply:
Maybe I should put it this way: as far as I know, it's always possible to calculate `KE` on the GPU. If we know of scenarios where it doesn't work, let's raise an issue!

Reply:
There's a bunch of tests here that all pass:

https://github.com/CliMA/Oceananigans.jl/blob/292da65c92d8b76f03cc95d88b5eb067e41e62a8/test/test_abstract_operations_computed_field.jl#L694-L733

Reply:
It does work on mine. But I have perceived some difference between GPUs. There are some operations that you report as not working for you that work for me and vice versa. So I think it may not work on all GPUs.

However, if you can think of an interesting example that won't compile on GPUs for me to substitute let me know and I'll change it! I honestly couldn't find any. Everything I've tried that was reasonable compiled successfully (kudos to you for significantly improving compilation!).

Reply:
I think there are examples on #1241 that are very long ?

Reply:
How about

```julia
julia> compute!(ComputedField(∂x(u)^2 + ∂y(v)^2 + ∂z(w)^2 + ∂x(w)^2))
ERROR: CUDA error: device kernel image is invalid (code 200, ERROR_INVALID_IMAGE)
```

Reply:
> So I think it may not work on all GPUs.

I remember you mentioning this but I think we need to carefully test that this is indeed the case before insinuating such in documentation --- because it does not seem possible except maybe for a machine-dependent bug in some low-level library.

Reply:
> I remember you mentioning this but I think we need to carefully test that this is indeed the case before insinuating such in documentation

I don't know what the state of affairs is right now after your improvements, but when I tested them before I was using the main branch (right after you posted your scripts) and I literally copy-pasted your scripts. So I don't know what room for error there is. But yes, I think we could at some point make a more rigorous test to see if everything is reproducible across all many GPUs as we can get.

Reply:
Switched the example to use energy dissipation rate. I think it's a better illustration and it'll definitely fails on GPUs.